src/Stack/Types/FlagName.hs

```diff
@@ -54,7 +54,7 @@ newtype FlagName =
   FlagName Text
   deriving (Typeable,Data,Generic,Hashable,Store,NFData)
 instance Eq FlagName where
-    x == y = compare x y == EQ
+    x == y = x == y
```

Comment:
It's funny that hlint would suggest this. :)

I'd feel safer with the old version